### DISCUSSION TITLE
Markdown file reached a size too large to render and view in repository

### DISCUSSION THREAD

**Discussion by b9Joker108** [community/community#110884] [Repositories]
### Select Topic Area

Question

### Body

Dear GitHub Community 

I started writing a personal log of my exploration of Bash and Zsh scripting in Termux within Notion, which is my most recent personal knowledge management solution, there have been others. The personal log grew and I felt that it could be usefully transformed into a book to inform others. The single page became too large for either the Notion Android app or the Notion Web-app to handle, and it started getting progressively glitchy. I managed the glitches in my stride and flow, until data and my edits were no longer preserved. It made me sad, as I lost a significant amount of work and the work was of standard. I then resolved to persue other solutions.

I had already created a GitHub account prior. And though I prospected other solutions to migrate the exported page from Notion, I ended up installing it into its own GitHub repository. Some information, such as images and graphemes were lost, but it was mostly a success. As a result, formatting of the migrated Notion Markdown file to GitHub Flavoured Markdown, even became more standardised, than in the Notion editing environment. I am currently using an Android tablet, and have found I cannot do nearly as much with the GitHub Android app, as I may do with the GitHub Web app. So, I mostly use the GitHub Android app with my smartphone and the GitHub Web-app with my tablet. 

The page or file, has grown significantly in its GitHub repository. I am really happy with the progress of the project and I feel it is a valuable body of work. I feel it will be of publishable quality and public interest. Moreover, I feel the work will be of significance to others when it is well-edited. Anyway, just recently, I must have past a size threshold for a Markdown file to render in a repository, and now it is not visible, and I cannot edit it in-state, within its repository via the GitHub Web-application. On investigation, I found I can continue editing it within the GitHub Android app or via the github.dev framework as different to GitHub Codespaces. I don't favour either of these editing documents for my tablet. github.dev has a number of shortcuts that I wish to use, but they are for Control or Function key combinations that do not work on my virtual keyboards within whichever Internet browser I use. I know I may clone the repository locally and then push changes to the GitHub repository via Termux, but I do not know if that is the best document solution to persue, as the file is of a significant size, book-length. 

I would really like to continue editing the document directly in the repository with the GitHub web-app, through an Internet browser. Is there a way that I may do that with large files? Or do you recommend another solution? I have found most Android apps not very useful when it comes to document production, and was really enjoying my experience writing in GitHub Flavoured Markdown in the GitHub repository. Is there any way I may make it work, even though the file is large? I am happy to pay a fair and reasonable price for the privilege? 

I await the community's advise,. 

Thanking you in anticipation 
Beauford

**Comment by MasterInQuestion** [comment]
Proper segmentation and hierarchy is crucial for accessibility.
    Does thoughtlessly tossing everything in 1 file really make sense..?

    “["github.dev"](https://github.dev/) has a number of shortcuts that I wish to use, but they are for Control or Function key combinations that do not work on my virtual keyboards within whichever Internet browser I use.”
<^>    Check this: [https://github<i>.</i>com/microsoft/vscode/issues/163432#issuecomment-1312029773](https://github.com/microsoft/vscode/issues/163432#issuecomment-1312029773)

**Reply by b9Joker108** [reply]
@MasterInQuestion I don't consider the way that I have so far progressed the work, thoughtless, but you are more than entitled to sport and exercise your opinion. It is not even at first blush, and is at rough pre-first draft status. The work evinces a structure, but as I have not yet resolved upon a definitive structure, this is somewhat makeshift, and is more the result of my creative process, than a conceptually planned and rigorously tight progression of the topic, subsuming tangentials. I feel, following other projects that I favour, that the work will at first edition stage, have a number of deliverables: Tarball, PDF, HTML, Markdown, Website, Wiki, for example, and the full set, is yet to be determined, and may also, in all of the formats, have three styles, complete-page, an archive by node, a conceptual tree. Somehow, so far unknown, with the oldschool print-forms, I need to generate an index of key-words, topics and concepts, tools, and a table-of-contents, both metatext/hyperlinks/hypertext. This all necessitates a process flow, I have not yet grappled with. But, I do contend, having the document primarily in a singular page format in the first instance, will likely make all of the mentioned process flows of the first edition deliverables, easier to realise in flow and process. If this is not the case, and my conjecture is ill-informed, I welcome advice, especially from those experienced in the process of the formation of such deliverables, in a state-of-the-art, best-practice fashion. I intend to write to GNU/FSF as to the nature of their process-flow, that they employ, to realise all of their deliverables, for example, the official Bash documentation, and there are others, I intent to model upon. If GitHub actually has a documented process on this, I would be replete with delight. I await your advice.

Also, I checked out the linkage you provided, and as I am using the Samsung Browser, which I understand is a proprietary fork of Chromium, though I may be in error, I don't really understand the reason for your posting of the linkage, for it is neither to do with github.dev, nor to do with the inability to do Control and Function key combinations, in virtual keyboards, within a tablet environment.

**Reply by MasterInQuestion** [reply]
Thoughtfully considered perhaps, thoughtlessly progressed though.
    Horrible one-liner, I reformatted it with rough parsing for readability:
[[
    I don't consider the way that I have so far progressed the work, thoughtless.
    But you are more than entitled to sport and exercise your opinion.

    It is not even at first blush, and is at rough pre-first draft status.
    The work evinces a structure, but as I have not yet resolved upon a definitive structure: this is somewhat makeshift.
    And is more the result of my creative process, than a conceptually planned and rigorously tight progression of the topic: subsuming tangentials.

    I feel, following other projects that I favour, that the work will at 1st edition stage, have a number of deliverables:
    Tarball, PDF, HTML, Markdown, Website, Wiki, for example; and the full set, is yet to be determined.
    And may also, in all of the formats, have 3 styles: complete-page, an archive by node, a conceptual tree.

    Somehow, so far unknown, with the oldschool print-forms:
    I need to generate an index of key-words, topics and concepts, tools, and a table-of-contents, both metatext/hyperlinks/hypertext.
    This all necessitates a process flow, I have not yet grappled with.

    But, I do contend, that:
    Having the document primarily in a single-page format in the initialization stage:
    Will likely make all the mentioned process flows of the 1st edition deliverables: easier to realise in flow and process.

    If this is not the case, be my conjecture ill-informed:
    I welcome advice, especially from those experienced in the process of formation of such deliverables: in a state-of-the-art, best-practice fashion.

    I intend to write to GNU/FSF as to the nature of their process-flow, that they employ, to realise all of their deliverables.
    For example, the official Bash documentation. And there are others I intent to model upon.

    If GitHub actually has a documented process on this, I would be replete with delight.
    I await your advice.
]]
    Why to bother so many formats: when things may be mostly clearly expressed in Plain-Text?
    Whatever, the problem is most likely that the document's hierarchy being ill-formed: the unreasonably conceived single-page.
    Essentially a issue pertaining to literacy.

    See also: https://github.com/orgs/community/discussions/110250#discussioncomment-8683398

[[
    Also, I checked the link you provided.
    And as I am using the Samsung Browser, which I understand is a proprietary fork of Chromium, though I may be in error.
    .
    I don't really understand the reason for your posting of the link:
    For it is neither to do with ["github.dev"](https://github.dev/); nor to do with the inability to do Control and Function key combinations, in virtual keyboards, within a tablet environment.
]]
    Trace further I shall suggest:
    https://docs.github.com/en/codespaces/the-githubdev-web-based-editor
    [https://github<i>.</i>com/microsoft/vscode/issues/163432#issuecomment-1344767469](https://github.com/microsoft/vscode/issues/163432#issuecomment-1344767469)
    [https://github<i>.</i>com/microsoft/vscode/issues/181874#issuecomment-1543416507](https://github.com/microsoft/vscode/issues/181874#issuecomment-1543416507)

**Reply by spenserblack** [reply]
Generally speaking I'd say you want to split up your document into multiple documents, especially considering the following (emphasis added)

> that the work will at first edition stage, have a number of deliverables: Tarball, PDF, HTML, *Markdown*, *Website*, *Wiki*

As a wiki or website you would most likely want to organize your documentation so that each Markdown file is a page. For other distributions, it's fairly common to organize the documentation so that each Markdown file is a chapter (I've stumbled on a few open-source books and this is what they do). I'd recommend looking into tools like [mdbook](https://github.com/rust-lang/mdBook) for guidance on how to organize your project and options for deploying/delivering it.

**Reply by b9Joker108** [reply]
@spenserblack  Your post is so welcome and useful. I didn't even know there was such a tool. I have installed mdbook with Termux. It is in the official Termux repository. I have also starred  the GitHub repository of the project. I have downloaded the mdbook documentation, for reference, and have glanced over it. I also had a look at gitbook which mdbook mentions and references, and which it is inspired by. Mdbook is a GitHub project and it is FOSS, which I favour. I also note, that gitbook is free for the use of single developers, and it integrates and automatically updates, and two-way syncs, with a GitHub archive. Gitbook also has AI integration, which is a significant plus. At this stage, as I am the only developer of the project, I may use gitbook, with view to potentially using mdbook in future, when more people are drawn to the project. I am so thankful for your thoughtful post.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @b9Joker108 ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. I've gone ahead and moved it for you. Good luck!

**Comment by b9Joker108** [comment]
@queenofcorgis Thank you so much for your action, in appropriately re-categorising my post. I will note the purpose of the 'Discussion' and 'Repositories' categories, going forward. I actually made a tentative request for two new functionalities, of the GitHub Web application, prior to this post, but no-one has yet responded. Would that likely be best-placed, in 'Discussions' or 'Repositories' or another category? B. T. W., I note that you are very community focussed, and involved in community building, I thank you so much for your endeavour. I have poked-my-nose into the Community section of GitHub a few times, mostly when I needed advice, and had a bit of an explore. I tend to be more focused on actively progressing and realising my projects, but I have resolved to take more active interest in the GitHub Community. I didn't know that the Community forum, was actually open to any GitHub member to actually comment upon the post of others, to assist them, but I read that yesterday, that this is favoured, when I was having a look-see around. So, I will be more active in the Community in future.

**Reply by queenofcorgis** [reply]
Thanks for the kind words and coming to the community, I hope you've gotten some good advice! And please - all GitHub users are welcome to participate - the more knowledge we all share the better we all can be 😃 . For feedback on the GitHub Web application - that would go in the [General category ](https://github.com/orgs/community/discussions/categories/general) - appreciate you asking.

**Comment by b9Joker108** [comment]
Mdbook and Gitbook are two solutions to the problem, so I have closed the issue.
